python-checks/src/main/java/org/sonar/python/checks/PandasReadNoDataTypeCheck.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6740")
+public class PandasReadNoDataTypeCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "The 'dtype' parameter should be used in calls to ";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, PandasReadNoDataTypeCheck::checkReadMethodCall);
+  }
+
+  private static void checkReadMethodCall(SubscriptionContext subscriptionContext) {
+    CallExpression callExpression = (CallExpression) subscriptionContext.syntaxNode();
+    Optional.of(callExpression)
+      .filter(PandasReadNoDataTypeCheck::isReadCall)
+      .map(CallExpression::arguments)
+      .filter(args -> TreeUtils.nthArgumentOrKeyword(1, "dtype", args) == null)
+      .ifPresent(args -> subscriptionContext.addIssue(callExpression.callee().lastToken(), getMessage(callExpression)));
```

Comment:
The reason is that I want the error to be raised only on the call to `read_csv` as opposed to the entire `pd.read_csv` expression. I hope this makes sense, but let me know if you think I should change it anyways.

Reply:
I believe the reporting would be slightly different, either highlighting the last component of the callee or the full callee (with qualifiers). 

I like the idea of only highlighting the last component of the callee. Now, using `lastToken()` may be a bit implicit in doing that, which may lead to reporting problems in edge cases (what if the callee is something else than a name or a qualified expression? e.g if the callee is `x[0]`, are we going to report on `]`? It's not going to happen because we wouldn't have found it to be a`read` call then, but this is a bit implicit).

Reply:
Thanks for bringing this up. I wasn't aware of the problems with just using lastToken. I have modified the way that I retrieve the `Name` in ada2452446f890659a100da728f2824e82cc5dbb, but these were done on top of changes introduced in 2e04a192739bf9eaa53e6dd621b93ed1d96d1576.